src/config.cpp

```diff
@@ -546,11 +546,55 @@ class SimulationConfig::Parser
             auto& valueIt = it.value();
             const auto debugStr = fmt::format("report {}", it.key());
             parseMandatory(valueIt, "cells", debugStr, report.cells);
+            parseOptional(valueIt, "sections", report.sections);
             parseMandatory(valueIt, "type", debugStr, report.type);
+            parseOptional(valueIt, "scaling", report.scaling);
+            parseOptional(valueIt, "compartments", report.compartments);
+            parseMandatory(valueIt, "variable_name", debugStr, report.variableName);
+            parseOptional(valueIt, "unit", report.unit);
             parseMandatory(valueIt, "dt", debugStr, report.dt);
             parseMandatory(valueIt, "start_time", debugStr, report.startTime);
             parseMandatory(valueIt, "end_time", debugStr, report.endTime);
             parseOptional(valueIt, "file_name", report.fileName);
+            parseOptional(valueIt, "enabled", report.enabled);
+
+            if (report.sections.empty()) {
+                report.sections = "soma";
+            } else {
+                std::set<std::string> possibleValues = {"soma", "axon", "dend", "apic", "all"};
+                if (possibleValues.find(report.sections) == possibleValues.end()) {
+                    throw SonataError(fmt::format("Section '{}' not supported ('{}') possible",
+                                                  report.sections,
+                                                  fmt::join(possibleValues, ", ")));
+                }
```

Comment:
Maybe we can have a function for this check ? It will be used in many places I guess.